### PR TITLE
Add frequency encoder

### PR DESCRIPTION
**Pull request by ppstacy**
Add frequency encoder code, update the test code and README.md correspondingly

### CODE REVIEW

File: kaggler/preprocessing/categorical.py [hunk 1]
```diff
@@ -547,3 +547,110 @@ def transform(self, X):
...
+
+        assert len(x) == len(y)
+
+        # NaN cannot be used as a key for dict. So replace it with a random
+        # integer
+        df = pd.DataFrame({y.name: y, x.name: x.fillna('NaN')})
+        df[x.name + '_freq'] = df[x.name].map(df[x.name].value_counts())
+        return df.groupby(x.name)[x.name + '_freq'].size().to_dict()
+
+    def fit(self, X, y):
```
Comment (jeongyoonlee) [inline kaggler/preprocessing/categorical.py]: Let's make `y` optional.

File: kaggler/preprocessing/categorical.py [hunk 2]
```diff
@@ -547,3 +547,110 @@ def transform(self, X):
...
+        """Initialize the FrequencyEncoder class object.
+        Args:
+            cv (sklearn.model_selection._BaseKFold, optional): sklearn CV object
+        """
+        self.cv = cv
+
+    def __repr__(self):
+        return('FrequencyEncoder(cv={})'.format(self.cv))
+
+    def _get_frequency_encoder(self, x, y):
```
Comment (jeongyoonlee) [inline kaggler/preprocessing/categorical.py]: We don't need/use `y` here. Let's remove it.

File: kaggler/preprocessing/categorical.py [hunk 3]
```diff
@@ -547,3 +547,110 @@ def transform(self, X):
...
+                    if i_enc == 1:
+                        x = X[col].fillna('NaN').map(frequency_encoder)
+                    else:
+                        x += X[col].fillna('NaN').map(frequency_encoder)
+
+                X.loc[:, col] = x / i_enc
+
+        return X
+
+    def fit_transform(self, X, y):
```
Comment (jeongyoonlee) [inline kaggler/preprocessing/categorical.py]: Same as above. Let's make `y` optional.

File: tests/test_encoders.py [hunk 4]
```diff
@@ -67,3 +67,28 @@ def test_EmbeddingEncoder(generate_data):
...
     assert X_emb.shape[1] == sum(ee.n_emb)
 
+
+def test_FrequencyEncoder(generate_data):
+    df = generate_data()
+    feature_cols = [x for x in df.columns if x != TARGET_COL]
+    cat_cols = [x for x in feature_cols if df[x].nunique() < 100]
+
+    te = FrequencyEncoder()
+    X_cat = te.fit_transform(df[cat_cols], df[TARGET_COL])
```
Comment (jeongyoonlee) [inline tests/test_encoders.py]: Remove `df[TARGET_COL]`.

File: tests/test_encoders.py [hunk 5]
```diff
@@ -67,3 +67,28 @@ def test_EmbeddingEncoder(generate_data):
...
+    print('Without CV:\n{}'.format(X_cat.head()))
+
+    assert X_cat.shape[1] == len(cat_cols)
+
+    cv = KFold(n_splits=N_FOLD, shuffle=True, random_state=RANDOM_SEED)
+    te = FrequencyEncoder(cv=cv)
+    X_cat = te.fit_transform(df[cat_cols], df[TARGET_COL])
```
Comment (jeongyoonlee) [inline tests/test_encoders.py]: Same as above.

File: tests/test_encoders.py [hunk 6]
```diff
@@ -67,3 +67,28 @@ def test_EmbeddingEncoder(generate_data):
...
+    print('With CV (fit_transform()):\n{}'.format(X_cat.head()))
+
+    assert X_cat.shape[1] == len(cat_cols)
+
+    te = FrequencyEncoder(cv=cv)
+    te.fit(df[cat_cols], df[TARGET_COL])
```
Comment (jeongyoonlee) [inline tests/test_encoders.py]: Same as above.

File: kaggler/preprocessing/categorical.py (hunk 2 above)
Comment (ppstacy) [reply kaggler/preprocessing/categorical.py]: yeah, thanks for pointing it out!

File: kaggler/preprocessing/categorical.py (hunk 1 above)
Comment (ppstacy) [reply kaggler/preprocessing/categorical.py]: done!

File: kaggler/preprocessing/categorical.py (hunk 3 above)
Comment (ppstacy) [reply kaggler/preprocessing/categorical.py]: done

File: tests/test_encoders.py (hunk 4 above)
Comment (ppstacy) [reply tests/test_encoders.py]: done

File: tests/test_encoders.py (hunk 5 above)
Comment (ppstacy) [reply tests/test_encoders.py]: done

File: tests/test_encoders.py (hunk 6 above)
Comment (ppstacy) [reply tests/test_encoders.py]: done

**Pull request closed without merging**